### PR TITLE
adding clarabel

### PR DESCRIPTION
**Pull request by goulart-paul**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by goulart-paul** [comment]
@conda-forge/help-python, ready for review!

File: recipes/clarabel/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{% set name = "clarabel" %}
+{% set version = "0.6.0" %}
+{% set checksum = "ef909a393e72981ca10b1d866d9cc7fb6295ece20ae035def764338894961184" %}
```
Comment (ocefpaf) [inline recipes/clarabel/meta.yaml:3]:
```suggestion
```

File: recipes/clarabel/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.python.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz  
+  sha256: {{ checksum }}
```
Comment (ocefpaf) [inline recipes/clarabel/meta.yaml:11]: No need for a variable that will be used only once. This just obfuscates the recipe.

```suggestion
  sha256: ef909a393e72981ca10b1d866d9cc7fb6295ece20ae035def764338894961184
```

File: recipes/clarabel/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    - crossenv                            # [build_platform != target_platform]                   
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
+  host:
+    - maturin
+    - python
+    - pip
+  run:
+    - python
+    - libgcc-ng  # [linux]
```
Comment (ocefpaf) [inline recipes/clarabel/meta.yaml:32]:
```suggestion
```

File: recipes/clarabel/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  skip: true  # [py<37]
+  script: 
+      - {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
+      - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+
+requirements:
+  build:
+    - python                              # [build_platform != target_platform]
+    - crossenv                            # [build_platform != target_platform]                   
+    - {{ compiler('rust') }}
```
Comment (ocefpaf) [inline recipes/clarabel/meta.yaml:24]: You don't need `libgcc-ng`, this is the "right" fix for it.

```suggestion
    - {{ compiler('c') }}
    - {{ compiler('rust') }}
```

Comment (goulart-paul) [reply recipes/clarabel/meta.yaml:24]: Thank you for the review comments.   And especially for this one, since I had no idea how to resolve that.

Comment (ocefpaf) [reply recipes/clarabel/meta.yaml:24]: IMO we should fix this, when a recipe requests `{{ compiler('rust') }}` everything should be in there to make it work. I'm not sure why we still need this. Maybe @isuruf can clarify?

Anyway, until we fix this, adding the c compiler is the right way to go.

Comment (synapticarbors) [reply recipes/clarabel/meta.yaml:24]: Do all rust packages additionally need the c compiler, or is it just a function of how pyo3 works in binding rust to the python c-api?

**Pull request merged**